vispy/visuals/volume.py

```diff
@@ -630,6 +611,21 @@ class VolumeVisual(Visual):
 
     _interpolation_names = ['linear', 'nearest']
 
+    _rendering_methods = {
+        'mip': MIP_SNIPPETS,
+        'minip': MINIP_SNIPPETS,
+        'attenuated_mip': ATTENUATED_MIP_SNIPPETS,
+        'iso': ISO_SNIPPETS,
+        'translucent': TRANSLUCENT_SNIPPETS,
+        'additive': ADDITIVE_SNIPPETS,
+        'average': AVG_SNIPPETS
+    }
+
+    _raycasting_modes = {
+        'volume': RAYCASTING_SETUP_VOLUME,
+        'plane': RAYCASTING_SETUP_PLANE
+    }
```

Comment:
I had a thought after seeing your updated tests. Should there maybe be a public property called `rendering_methods` which are the names that users can choose from? Could be simply `list(self._rendering_methods.keys())`. Same for raycasting modes. That way UI applications (like napari or others) could have public access to something that lists the choices.

Although, that doesn't make the related test(s) any cleaner as this property would be an instance attribute. Any clean way to make it a class attribute?

Reply:
Really like this - my instinct (very much napari biased) is to make this an enum at the module level in a volume_constants.py which is accessible as a class attribute on the visual, ideally with the property syntax

It looks like 'classproperty' can be rigged up pretty simply - cool! https://stackoverflow.com/questions/5189699/how-to-make-a-class-property

Here is an example in napari (where we should probably also rig up 'classproperty') https://github.com/napari/napari/blob/main/napari/layers/image/_image_constants.py

Reply:
It could be done simpler, but more error prone, by just doing `raycasting_modes = list(_raycasting_modes.keys())` right after the dictionary declaration. Then it is just up to the subclass to make sure the dict and the list both get updated.

How does using an enum effect people adding their own raycasting modes? You can't update an enum can you? An enum kind of feels like overkill here.

Reply:
Yeah, I also was thinking of a class property! I don't feel strongly about enums over lists, but I think it should be up to date with the subclass without forcing the subclasser to deal with it.